src/testController.ts

```diff
@@ -0,0 +1,207 @@
+import { exec } from "child_process";
+import { promisify } from "util";
+
+import * as vscode from "vscode";
+import { CodeLens } from "vscode-languageclient/node";
+
+import { Ruby } from "./ruby";
+import { Command } from "./status";
+
+const asyncExec = promisify(exec);
+
+export class TestController {
+  private testController: vscode.TestController;
+  private testCommands: WeakMap<vscode.TestItem, string>;
+  private testRunProfile: vscode.TestRunProfile;
+  private workingFolder: string;
+  private ruby: Ruby;
+
+  constructor(
+    context: vscode.ExtensionContext,
+    workingFolder: string,
+    ruby: Ruby
+  ) {
+    this.workingFolder = workingFolder;
+    this.ruby = ruby;
+
+    this.testController = vscode.tests.createTestController(
+      "rubyTests",
+      "Ruby Tests"
+    );
+
+    this.testCommands = new WeakMap<vscode.TestItem, string>();
+
+    this.testRunProfile = this.testController.createRunProfile(
+      "Run",
+      vscode.TestRunProfileKind.Run,
+      (request, token) => {
+        this.runHandler(request, token);
+      },
+      true
+    );
+
+    vscode.commands.executeCommand("testing.clearTestResults");
```

Comment:
I added in this line to clear the test results each time the LSP is started/restarted.

BTW, `testing.clearTestResults` and `testing.showMostRecentOutput` are not in the VS Code API documentation (or I couldn't find them in there). I got these commands from the source code [here](https://github.com/microsoft/vscode/blob/8f74fbfd1f2d8f6268a42df131726b218aafe511/src/vs/workbench/contrib/testing/common/constants.ts#L53). There are other test-related commands that might be useful later so I wanted to document it.